python/pyspark/tests.py

```diff
@@ -618,10 +618,13 @@ def test_barrier_with_python_worker_reuse(self):
         """
         Verify that BarrierTaskContext.barrier() with reused python worker.
         """
+        self.sc._conf.set("spark.python.work.reuse", "true")
```

Comment:
@xuanyuanking, this will probably need a separate suite case since it's also related with how we start the worker or not. You can make a new class, run a simple job to make sure workers are created and being resued, test it and stop.

Reply:
I do these 2 check like below:
1. Run this test case without fix in `BarrierTaskContext._getOrCreate`, the bug can be reproduced.
2. Same code running in pyspark shell and set `spark.python.work.resue=false`, it return successfully.
Maybe this can prove the UT can cover the issue and also can reuse the original barrier case code, WDYT?

Reply:
@HyukjinKwon Hi Hyukjin if you still think this need a separate class I'll think about the method of checking worker reuse and give a follow up PR.

Reply:
Yup. sorry for late response.